libbzip2-rs-sys/src/bzlib.rs

```diff
@@ -737,32 +753,25 @@ fn copy_input_until_stop(strm: &mut bz_stream, s: &mut EState) -> bool {
     progress_in
 }
 
-unsafe fn copy_output_until_stop(strm: &mut bz_stream, s: &mut EState) -> bool {
+fn copy_output_until_stop(strm: &mut bz_stream, s: &mut EState) -> bool {
     let mut progress_out = false;
 
     let zbits = &mut s.arr2.raw_block()[s.nblock as usize..];
 
     loop {
-        if strm.avail_out == 0 {
+        if s.state_out_pos >= s.writer.num_z as i32 {
             break;
         }
-        if s.state_out_pos >= s.writer.num_z as i32 {
+        if !strm.write_byte(zbits[s.state_out_pos as usize]) {
             break;
```

Comment:
could this use `if let Some(v) = zbits.get(s.state_out_pos)`?

Reply:
`num_z` can be less than `zbits.len()`, but never bigger. As such I think we should keep panicking if `zbits[s.state_out_pos]` is out of bounds.

Reply:
huh? doesn't the top check verify that the index is in-range? In what case could this panic?